src/Expression/ExpressionBuilder.php

```diff
@@ -14,12 +21,178 @@
  *
  * These expressions can be used with the query builder to build complex and customizable database queries
  * {@see Expression} class.
+ *
+ * @psalm-import-type ParamsType from ConnectionInterface
  */
 class ExpressionBuilder implements ExpressionBuilderInterface
 {
+    public function __construct(private QueryBuilderInterface|null $queryBuilder = null)
```

Comment:
Why is query builder optional?

Reply:
For BC. In the next maror version `null` can be removed.